src/http2_connection.erl

```diff
@@ -778,111 +822,43 @@ route_frame(Frame, #connection{}=Conn) ->
-    %% make another decision
-
-    %% If it was MAX_FRAME_SIZE, then we recurse into this same
-    %% function, because we're able to send another frame of some
-    %% length.
-
-    %% If it was connection send window size, we're blocked at the
-    %% connection level and we should break out of this recursion
-
-    %% If it was stream send_window size, we're blocked on this
-    %% stream, but other streams can still go, so we'll break out of
-    %% this recursion, but not the connection level
-
-    SSWS = Stream#stream.send_window_size,
-
-    QueueSize = byte_size(Stream#stream.queued_data),
-
-    {MaxToSend, ExitStrategy} =
-        case {MFS =< SWS andalso MFS =< SSWS, SWS < SSWS} of
-            %% If MAX_FRAME_SIZE is the smallest, send one and recurse
-            {true, _} ->
-                {MFS, max_frame_size};
-            {false, true} ->
-                {SWS, connection};
-            _ ->
-                {SSWS, stream}
-        end,
-
-    {Frame, SentBytes, NewS} =
-        case MaxToSend > QueueSize of
-            true ->
-                Flags = case Stream#stream.body_complete of
-                         true -> ?FLAG_END_STREAM;
-                         false -> 0
-                        end,
-                %% We have the power to send everything
-                {{#frame_header{
-                     stream_id=Stream#stream.id,
-                     flags=Flags,
-                     type=?DATA,
-                     length=QueueSize
-                    },
-                  http2_frame_data:new(Stream#stream.queued_data)}, %% Full Body
-                 QueueSize,
-                 Stream#stream{
-                   queued_data=done,
-                   send_window_size=SSWS-QueueSize}};
-            false ->
-                <<BinToSend:MaxToSend/binary,Rest/binary>> = Stream#stream.queued_data,
-                {{#frame_header{
-                     stream_id=Stream#stream.id,
-                     type=?DATA,
-                     length=MaxToSend
-                    },
-                  http2_frame_data:new(BinToSend)},
-                 MaxToSend,
-                 Stream#stream{
-                   queued_data=Rest,
-                   send_window_size=SSWS-MaxToSend}}
-        end,
-
-    _Sent = http2_stream:send_frame(Stream#stream.pid, Frame),
-
-    case ExitStrategy of
-        max_frame_size ->
-            s_send_what_we_can(SWS - SentBytes, MFS, NewS);
-        stream ->
-            {SWS - SentBytes, NewS};
-        connection ->
-            {0, NewS}
-    end.
+handle_event({stream_finished,
+              StreamId,
+              Headers,
+              Body}, StateName,
+             Conn) ->
+    Stream = h2_stream_set:get(StreamId, Conn#connection.streams),
+    case h2_stream_set:type(Stream) of
+        active ->
+            NotifyPid = h2_stream_set:notify_pid(Stream),
+            Response =
+                case Conn#connection.type of
+                    server -> garbage;
```

Comment:
@stillinbeta here is where we automatically set response content to `garbage` on the server side